File: js/linkList.js
function Node(value) {
    return {
      value: value,
      next: null,
    };
  }
  
  function LinkedList() {
    const list = {
      head: null,
      tail: null,
      length: 0,
    };
  
    function add(value) {
      const node = Node(value);
      if (list.head === null) {
        list.head = node;
        list.tail = node;
      } else {
        list.tail.next = node;
        list.tail = node;
      }
      list.length++;
    }
  
    function insert(value, index) {
      if (index < 0 || index > list.length) {
        throw new Error('Index out of range');
      }
  
      const node = Node(value);
      if (index === 0) {
        node.next = list.head;
        list.head = node;
        if (list.tail === null) {
          list.tail = node;
        }
      } else if (index === list.length) {
        list.tail.next = node;
        list.tail = node;
      } else {
        let current = list.head;
        for (let i = 0; i < index - 1; i++) {
          current = current.next;
        }
        node.next = current.next;
        current.next = node;
      }
      list.length++;
    }
  
    function search(value) {
      let current = list.head;
      while (current !== null) {
        if (current.value === value) {
          return true;
        }
        current = current.next;
      }
      return false;
    }
  
    function remove(value) {
      if (list.head === null) {
        return;
      }
      if (list.head.value === value) {
        list.head = list.head.next;
        if (list.head === null) {
          list.tail = null;
        }
        list.length--;
        return;
      }
      let current = list.head;
      while (current.next !== null && current.next.value !== value) {
        current = current.next;
      }
      if (current.next !== null) {
        current.next = current.next.next;
        if (current.next === null) {
          list.tail = current;
        }
        list.length--;
      }
    }
  
    function sort() {
      const result = [];
      let current = list.head;
      while (current !== null) {
        result.push(current.value);
        current = current.next;
      }
      result.sort((a, b) => a - b);
      return result;
    }
  
    return {
      add,
      insert,
      search,
      remove,
      sort,
    };
  }
  const linkedList = LinkedList();

  linkedList.add(5);
  linkedList.add(3);
  linkedList.add(7);
  linkedList.add(1);
  linkedList.add(9);
  
  console.log(linkedList.sort()); 
  
  linkedList.remove(7);
  
  console.log(linkedList.search(7)); 
  
  linkedList.insert(6, 2);
  
  console.log(linkedList.sort());
  

File: Python/linkList.py
class Node:
    def __init__(self, data=None):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None

    def add(self, data):
        new_node = Node(data)
        if self.head is None:
            self.head = new_node
        else:
            current = self.head
            while current.next is not None:
                current = current.next
            current.next = new_node

    def remove(self, data):
        if self.head is None:
            return
        if self.head.data == data:
            self.head = self.head.next
            return
        current = self.head
        while current.next is not None:
            if current.next.data == data:
                current.next = current.next.next
                return
            current = current.next

    def insert(self, data, position):
        new_node = Node(data)
        if position == 0:
            new_node.next = self.head
            self.head = new_node
            return
        current = self.head
        for i in range(position - 1):
            if current is None:
                return
            current = current.next
        new_node.next = current.next
        current.next = new_node

    def search(self, data):
        current = self.head
        while current is not None:
            if current.data == data:
                return True
            current = current.next
        return False

    def sort(self):
        if self.head is None:
            return
        current = self.head
        while current.next is not None:
            next_node = current.next
            while next_node is not None:
                if current.data > next_node.data:
                    temp = current.data
                    current.data = next_node.data
                    next_node.data = temp
                next_node = next_node.next
            current = current.next

    def __str__(self):
        if self.head is None:
            return "Empty List"
        current = self.head
        nodes = []
        while current is not None:
            nodes.append(str(current.data))
            current = current.next
        return "->".join(nodes)
linked_list = LinkedList()
linked_list.add(3)
linked_list.add(1)
linked_list.add(2)
print(linked_list)  
linked_list.sort()
print(linked_list)  
linked_list.remove(2)
print(linked_list)  
linked_list.insert(2, 1)
print(linked_list)  
print(linked_list.search(2))  
print(linked_list.search(4))  